### ISSUE TITLE
Can not start dockerd, getting Error starting daemon: Error initializing network controller: Error creating default network: HNS failed with error : Unspecified error

### ISSUE THREAD

**Issue by artisticcheese** [moby/moby#34018]
**Description**
Can not start dockerd on Windows 2016 1607 (14393.1358), getting error 
`
time="2017-07-07T13:45:42.799250600-06:00" level=debug msg="start clean shutdown of all containers with a 15 seconds tim
eout..."
Error starting daemon: Error initializing network controller: Error creating default network: HNS failed with error : Unspecified error`

**Steps to reproduce the issue:**
1. Start dockerd
2.
3.

**Describe the results you received:**
Does not start

**Additional information you deem important (e.g. issue happens only occasionally):**
Happens all the time. Here is output of `dockerd --debug`. I start `dockerd` with following switches then it starts fine `dockerd --debug --bridge none`
```
time="2017-07-07T13:45:42-06:00" level=warning msg="The \"graph\" config file option is deprecated. Please use \"data-ro
ot\" instead."
time="2017-07-07T13:45:42.695223700-06:00" level=debug msg="Listener created for HTTP on npipe (//./pipe/docker_engine)"

time="2017-07-07T13:45:42.698224300-06:00" level=debug msg="Stackdump - waiting signal at Global\\docker-daemon-4572"
time="2017-07-07T13:45:42.698224300-06:00" level=info msg="Windows default isolation mode: process"
time="2017-07-07T13:45:42.699225100-06:00" level=debug msg="Using default logging driver json-file"
time="2017-07-07T13:45:42.700225000-06:00" level=debug msg="WindowsGraphDriver InitFilter at E:\\images\\windowsfilter"
time="2017-07-07T13:45:42.704226900-06:00" level=info msg="[graphdriver] using prior storage driver: windowsfilter"
time="2017-07-07T13:45:42.705228900-06:00" level=debug msg="Using graph driver windowsfilter"
time="2017-07-07T13:45:42.721230500-06:00" level=debug msg="Max Concurrent Downloads: 3"
time="2017-07-07T13:45:42.721230500-06:00" level=debug msg="Max Concurrent Uploads: 5"
time="2017-07-07T13:45:42.734234600-06:00" level=info msg="Graph migration to content-addressability took 0.00 seconds"
time="2017-07-07T13:45:42.734234600-06:00" level=info msg="Loading containers: start."
time="2017-07-07T13:45:42.736237900-06:00" level=debug msg="Option Experimental: false"
time="2017-07-07T13:45:42.737236900-06:00" level=debug msg="Option DefaultDriver: nat"
time="2017-07-07T13:45:42.737236900-06:00" level=debug msg="Option DefaultNetwork: nat"
time="2017-07-07T13:45:42.738234500-06:00" level=info msg="Restoring existing overlay networks from HNS into docker"
time="2017-07-07T13:45:42.739235900-06:00" level=debug msg="[GET]=>[/networks/] Request : "
time="2017-07-07T13:45:42.749237700-06:00" level=debug msg="Network Response : [  ]"
time="2017-07-07T13:45:42.766242200-06:00" level=debug msg="[GET]=>[/networks/] Request : "
time="2017-07-07T13:45:42.771246500-06:00" level=debug msg="Network Response : [  ]"
time="2017-07-07T13:45:42.772243500-06:00" level=debug msg="Launching DNS server for network%!(EXTRA string=none)"
time="2017-07-07T13:45:42.774244900-06:00" level=debug msg="Allocating IPv4 pools for network nat (85aa34f5bdfe1a80135f8
3b42e6f87e6850d2802ee4e0688b075a9253308301d)"
time="2017-07-07T13:45:42.774244900-06:00" level=debug msg="RequestPool(LocalDefault, , , map[], false)"
time="2017-07-07T13:45:42.775244700-06:00" level=debug msg="RequestAddress(0.0.0.0/0, <nil>, map[RequestAddressType:com.
docker.network.gateway])"
time="2017-07-07T13:45:42.776244900-06:00" level=debug msg="HNSNetwork Request ={\"Name\":\"nat\",\"Type\":\"nat\",\"Sub
nets\":[{\"AddressPrefix\":\"0.0.0.0/0\",\"GatewayAddress\":\"0.0.0.0\"}]} Address Space=[{0.0.0.0/0 0.0.0.0 []}]"
time="2017-07-07T13:45:42.777245400-06:00" level=debug msg="[POST]=>[/networks/] Request : {\"Name\":\"nat\",\"Type\":\"
nat\",\"Subnets\":[{\"AddressPrefix\":\"0.0.0.0/0\",\"GatewayAddress\":\"0.0.0.0\"}]}"
time="2017-07-07T13:45:42.796249200-06:00" level=debug msg="releasing IPv4 pools from network nat (85aa34f5bdfe1a80135f8
3b42e6f87e6850d2802ee4e0688b075a9253308301d)"
time="2017-07-07T13:45:42.796249200-06:00" level=debug msg="ReleaseAddress(0.0.0.0/0, 0.0.0.0)"
time="2017-07-07T13:45:42.798252000-06:00" level=debug msg="ReleasePool(0.0.0.0/0)"
time="2017-07-07T13:45:42.799250600-06:00" level=debug msg="start clean shutdown of all containers with a 15 seconds tim
eout..."
Error starting daemon: Error initializing network controller: Error creating default network: HNS failed with error : Un
specified error
```
**Output of `docker version`:**

```
Client:
 Version:      17.06.0-ce
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   02c1d87
 Built:        Fri Jun 23 21:30:30 2017
 OS/Arch:      windows/amd64
error during connect: Get http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.30/version: open //./pipe/docker_engine: The system c
annot find the file specified. In the default daemon configuration on Windows, the docker client must be run elevated to
 connect. This error may also indicate that the docker daemon is not running.
```

**Output of `docker info`:**

```
Does not work since dockerd is not running
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by artisticcheese** [comment]
If anybody have the same issue. You need to stop hns service `Stop-Service hns` delete file from `C:\ProgramData\Microsoft\Windows\HNS\hns.data` and then start hns service.

**Comment by friism** [comment]
I haven't tested, but I guess the clean-up script could work too: https://github.com/Microsoft/Virtualization-Documentation/tree/master/windows-server-container-tools/CleanupContainerHostNetworking

cc @kallie-b

**Comment by artisticcheese** [comment]
Tried clean up script but it id not delete file I mentioned.

**Comment by friism** [comment]
@artisticcheese not even with the -force param?

**Comment by artisticcheese** [comment]
I run PS1 file with `-Cleanup -ForceDeleteAllSwitches` switches and this did not fix the issue untill I manually removed the file. Script did not fail but did not complaint about anything either.

**Comment by friism** [comment]
@kallie-b should that be part of the script?

**Comment by kallie-b** [comment]
@artisticcheese, @friism 

Yes, right now the script depends on Docker running to work as expected. I assume that since Docker wasn't running, the script didn't work properly. I'll investigate this and put it on my TODO list to update the script.

**Comment by hanxinimm** [comment]
I have the same error

System: Server 2016  Standard

PS C:\Users\Administrator> dockerd --debug
time="2017-09-13T20:44:50.917765400+08:00" level=debug msg="Listener created for HTTP on npipe (//./pipe/docker_engine)" 
time="2017-09-13T20:44:50.953814600+08:00" level=info msg="Windows default isolation mode: process" 
time="2017-09-13T20:44:50.953814600+08:00" level=debug msg="Using default logging driver json-file" 
time="2017-09-13T20:44:50.953814600+08:00" level=debug msg="Stackdump - waiting signal at Global\\docker-daemon-8868" 
time="2017-09-13T20:44:50.954792200+08:00" level=debug msg="WindowsGraphDriver InitFilter at C:\\ProgramData\\docker\\windowsfilter" 
time="2017-09-13T20:44:50.954792200+08:00" level=debug msg="Using graph driver windowsfilter" 
time="2017-09-13T20:44:50.954792200+08:00" level=debug msg="Max Concurrent Downloads: 3" 
time="2017-09-13T20:44:50.954792200+08:00" level=debug msg="Max Concurrent Uploads: 5" 
time="2017-09-13T20:44:50.961795500+08:00" level=info msg="Graph migration to content-addressability took 0.00 seconds" 
time="2017-09-13T20:44:50.961795500+08:00" level=info msg="Loading containers: start." 
time="2017-09-13T20:44:50.961795500+08:00" level=debug msg="Option Experimental: false" 
time="2017-09-13T20:44:50.961795500+08:00" level=debug msg="Option DefaultDriver: nat" 
time="2017-09-13T20:44:50.961795500+08:00" level=debug msg="Option DefaultNetwork: nat" 
time="2017-09-13T20:44:50.962794800+08:00" level=info msg="Restoring existing overlay networks from HNS into docker" 
time="2017-09-13T20:44:50.962794800+08:00" level=debug msg="[GET]=>[/networks/] Request : " 
time="2017-09-13T20:44:50.983806900+08:00" level=debug msg="Network Response : [  ]" 
time="2017-09-13T20:44:51.042854200+08:00" level=debug msg="[GET]=>[/networks/] Request : " 
time="2017-09-13T20:44:51.055863500+08:00" level=debug msg="Network Response : [  ]" 
time="2017-09-13T20:44:51.058866500+08:00" level=debug msg="Launching DNS server for network%!(EXTRA string=none)" 
time="2017-09-13T20:44:51.065871100+08:00" level=debug msg="Allocating IPv4 pools for network nat (2555fc6b3c4a16f56af25749be3d90dc43ff977209a6130adf885f1ba98b295b)" 
time="2017-09-13T20:44:51.065871100+08:00" level=debug msg="RequestPool(LocalDefault, , , map[], false)" 
time="2017-09-13T20:44:51.065871100+08:00" level=debug msg="RequestAddress(0.0.0.0/0, <nil>, map[RequestAddressType:com.docker.network.gateway])" 
time="2017-09-13T20:44:51.065871100+08:00" level=debug msg="HNSNetwork Request ={\"Name\":\"nat\",\"Type\":\"nat\",\"Subnets\":[{\"AddressPrefix\":\"0.0.0.0/0\",\"GatewayAddress\":\"0.0.0.0\"}]} Address Sp
ace=[{0.0.0.0/0 0.0.0.0 []}]" 
time="2017-09-13T20:44:51.065871100+08:00" level=debug msg="[POST]=>[/networks/] Request : {\"Name\":\"nat\",\"Type\":\"nat\",\"Subnets\":[{\"AddressPrefix\":\"0.0.0.0/0\",\"GatewayAddress\":\"0.0.0.0\"}]}
" 
time="2017-09-13T20:44:51.968505700+08:00" level=debug msg="releasing IPv4 pools from network nat (2555fc6b3c4a16f56af25749be3d90dc43ff977209a6130adf885f1ba98b295b)" 
time="2017-09-13T20:44:51.968505700+08:00" level=debug msg="ReleaseAddress(0.0.0.0/0, 0.0.0.0)" 
time="2017-09-13T20:44:51.968505700+08:00" level=debug msg="ReleasePool(0.0.0.0/0)" 
time="2017-09-13T20:44:51.968505700+08:00" level=debug msg="start clean shutdown of all containers with a 15 seconds timeout..." 
dockerd : Error starting daemon: Error initializing network controller: Error creating default network: HNS failed with error : Unspecified error
+ dockerd --debug
+ ~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (Error starting ...specified error:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError

delete file from C:\ProgramData\Microsoft\Windows\HNS\hns.data  not fix it

**Comment by sampathmende** [comment]
Is there any update on this issue. I am facing same issue . I clicked docker for windows icon and but showing same above error. system details : 
OS: Windows 10 Pro
Edition: Professional
Id: 1709
Build: 16299
BuildLabName: 16299.15.amd64fre.rs3_release.170928-1534

Docker community verions
Version 17.12.0-ce-win46 (15048)
Channel: stable
0ac7325

Issue description ::
09:39:55.939][WindowsDaemon  ][Error  ] fatal: Error starting daemon: Error initializing network controller: Error creating default network: HNS failed with error : Unspecified error
msg="open \\\\.\\pipe\\docker_engine_windows: The system cannot find the file specified."
[09:39:55.946][DockerDaemonChecker][Error  ] Docker daemon is not running
[09:39:57.020][Cmd            ][Info   ] time="2018-01-11T09:39:57+05:30" level=fatal msg="The specified service already exists."
I did delete the below file and started docker many times but not resolved.
C:\ProgramData\Microsoft\Windows\HNS\hns.data
Pls help as i am a beginner.
